### DISCUSSION TITLE
Must `tbl` be used in `where`?

### DISCUSSION THREAD

**Discussion by ddxl123** [simolus3/drift#2052] [Q&A] [answered]
I already knew it:
```
select( tableA ).join([ innerJoin( tableB, tableB.aId.equalsExp( tableA .id ) ),],)
      ..where( tableB.id.equals( bId ));
```

but:
```
select( tableA )
      ..join([ innerJoin( tableB, tableB.aId.equalsExp( tableA .id ) ),],)
      ..where((tbl) => tableB.id.equals( bId ));
```
`where` uses `tableB.id.equals()`, not `tbl.id.equals()` , is this OK?

I think the second way of writing it looks more comfortable when there is a join.

**Comment by simolus3** [comment, marked as answer]
This is essentially the same question as https://github.com/simolus3/drift/discussions/2055, right?

When you call something with a cascade operator (`..`), it returns the original value. So `select(tableA)..join(...)` __does not__ give you a `JoinedSelectStatement`, it stays a single-table `SimpleSelectStatement`.
When you use `select(tableA).join(...)`, you get the result of the `join` call which is the `JoinedSelectStatement` that you're interested in.

By the way, all five discussions you've opened essentially resolve around the same topic. You would save both of us a lot of time by just asking that question in a single discussion and perhaps wait for responses in there before opening very similar questions multiple time.
